eth1_refactor/eventdispatcher/event_dispatcher.go

```diff
@@ -0,0 +1,74 @@
+package eventdispatcher
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+)
+
```

Comment:
I'd move all the interfaces above to here, having them in separate files doesn't really feel clean. And in practice they are used here, so it makes sense to have them here.